### ISSUE TITLE
[Bug]: Screen reader doesn't read required field for Label

### ISSUE THREAD

**Issue by anmohanty123** [microsoft/fluentui#23045]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 10 10.0.22000
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz
    Memory: 13.83 GB / 31.71 GB
  Browsers:
    Edge: Spartan (44.22000.120.0), Chromium (101.0.1210.47)
    Internet Explorer: 11.0.22000.120
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://codepen.io/anmohanty123/pen/WNMpMwZ

## Actual Behavior

Screen reader does not read required field for Label

## Expected Behavior
Screen reader should read required field for Label

### Requested priority

Low

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by gouttierre** [comment]
@anmohanty123 - Thanks for filing this issue with us. I can confirm screen reader announces' I'm a label' without the required attribute. To set expectations the developers will review this issue once capacity allows.

@khmakoto - Would you be able to confirm if this is a regression, or if this behavior is an issue? I have added it to the a11y bucket.
cc @smhigley
